### PR TITLE
Remove map_can_construct[_with_clear]_at() and gMapGroundFlags

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/actions/RideEntranceExitPlaceAction.cpp [hunk 1]
```diff
@@ -99,15 +99,15 @@ GameActions::Result::Ptr RideEntranceExitPlaceAction::Query() const
     }
 
     auto clear_z = z + (_isExit ? RideExitHeight : RideEntranceHeight);
-    auto cost = MONEY32_UNDEFINED;
-    if (!map_can_construct_with_clear_at(
-            { _loc, z, clear_z }, &map_place_non_scenery_clear_func, { 0b1111, 0 }, GetFlags(), &cost,
-            CREATE_CROSSING_MODE_NONE))
+    auto canBuild = MapCanConstructWithClearAt(
+        { _loc, z, clear_z }, &map_place_non_scenery_clear_func, { 0b1111, 0 }, GetFlags() | GAME_COMMAND_FLAG_APPLY);
```
Comment (Gymnasiast) [inline src/openrct2/actions/RideEntranceExitPlaceAction.cpp:103]:
```suggestion
        { _loc, z, clear_z }, &map_place_non_scenery_clear_func, { 0b1111, 0 }, GetFlags());
```

File: src/openrct2/actions/FootpathPlaceAction.cpp [hunk 2]
```diff
@@ -248,16 +248,17 @@ GameActions::Result::Ptr FootpathPlaceAction::ElementInsertQuery(GameActions::Re
...
+    auto canBuild = MapCanConstructWithClearAt(
+        { _loc, zLow, zHigh }, &map_place_non_scenery_clear_func, quarterTile, GetFlags(), crossingMode);
+    if (!entrancePath && canBuild->Error != GameActions::Status::Ok)
     {
-        return MakeResult(
-            GameActions::Status::NoClearance, STR_CANT_BUILD_FOOTPATH_HERE, gGameCommandErrorText, gCommonFormatArgs);
+        canBuild->ErrorTitle = STR_CANT_BUILD_FOOTPATH_HERE;
+        return canBuild;
     }
+    res->Cost += canBuild->Cost;
```
Comment (duncanspumpkin) [inline src/openrct2/actions/FootpathPlaceAction.cpp:258]: This wasnt there before. We will need to increment network version.

File: src/openrct2/actions/FootpathPlaceFromTrackAction.cpp [hunk 3]
```diff
@@ -199,17 +199,16 @@ GameActions::Result::Ptr FootpathPlaceFromTrackAction::ElementInsertExecute(Game
...
     {
-        return MakeResult(
-            GameActions::Status::NoClearance, STR_RIDE_CONSTRUCTION_CANT_CONSTRUCT_THIS_HERE, gGameCommandErrorText,
-            gCommonFormatArgs);
+        canBuild->ErrorTitle = STR_RIDE_CONSTRUCTION_CANT_CONSTRUCT_THIS_HERE;
+        return canBuild;
     }
-
-    gFootpathGroundFlags = gMapGroundFlags;
+    res->Cost += canBuild->Cost;
```
Comment (duncanspumpkin) [inline src/openrct2/actions/FootpathPlaceFromTrackAction.cpp:210]: This on the other hand is in the execute.

**Review by duncanspumpkin — APPROVED** [review]

File: src/openrct2/actions/RideEntranceExitPlaceAction.cpp [hunk 4]
```diff
@@ -99,15 +99,15 @@ GameActions::Result::Ptr RideEntranceExitPlaceAction::Query() const
     }
 
     auto clear_z = z + (_isExit ? RideExitHeight : RideEntranceHeight);
-    auto cost = MONEY32_UNDEFINED;
```
Comment (duncanspumpkin) [inline src/openrct2/actions/RideEntranceExitPlaceAction.cpp:102]: accidently removed costs of removing casuses #17904

File: src/openrct2/actions/RideEntranceExitPlaceAction.cpp [hunk 5]
```diff
@@ -163,12 +163,12 @@ GameActions::Result::Ptr RideEntranceExitPlaceAction::Execute() const
...
```
Comment (duncanspumpkin) [inline src/openrct2/actions/RideEntranceExitPlaceAction.cpp:166]: accidently removed costs of removing casuses #17904

File: src/openrct2/actions/RideEntranceExitPlaceAction.cpp [hunk 6]
```diff
@@ -229,14 +229,14 @@ GameActions::Result::Ptr RideEntranceExitPlaceAction::TrackPlaceQuery(const Coor
 
     int16_t baseZ = loc.z;
     int16_t clearZ = baseZ + (isExit ? RideExitHeight : RideEntranceHeight);
-    auto cost = MONEY32_UNDEFINED;
```
Comment (duncanspumpkin) [inline src/openrct2/actions/RideEntranceExitPlaceAction.cpp:232]: and here

**Pull request merged**